### PR TITLE
fix skin_logo config, again

### PR DESCRIPTION
**Pull request by johndoh**
I apologise for how much trouble the skin_logo config option has been since i first modified it.

I've fixed the bug raised in #7456 by adding a new param which accounts for the special case of the print icon and allows for similar cases in future. I've also added a bunch more tests to make sure all the different uses are properly covered.

### CODE REVIEW

File: skins/elastic/templates/includes/menu.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <div id="layout-menu">
 	<h2 id="aria-label-tasknav" class="voice"><roundcube:label name="arialabeltasknav" /></h2>
 	<div class="popover-header">
-		<roundcube:object name="logo" src="/images/logo.svg" data-src-small="0" id="logo" alt="Logo" />
+		<roundcube:object name="logo" src="/images/logo.svg" alt-logos="dark,small,small-dark" id="logo" alt="Logo" />
```
Comment (alecpl) [inline skins/elastic/templates/includes/menu.html:4]: Do we need `alt-logos` attribute at all?

Comment (johndoh) [reply skins/elastic/templates/includes/menu.html:4]: TBH I'm not sure. I thought this way allowed skins to define which alt types they support.

Comment (johndoh) [reply skins/elastic/templates/includes/menu.html:4]: Alternatively I could add something like `supported_logo_types` in meta.json and not have this attribute

**Comment by alecpl** [comment]
I'd prefer to have the watermark thing separated. I'm not sure this is the best way to go with watermark images. Maybe we should rather make the watermark.html a template (with normal img element) parsed by the templates engine.

**Comment by gianlucagiacometti** [comment]
I agree with @alecpl on the watermark.
At the moment the best way I found to customise it is to rewrite in a temp dir the watermark.html with the customised image and change the env var watermark accordingly. Having the image directly customisable in the core would be wonderful.

**Comment by johndoh** [comment]
In the past Alec has said he didn't want to move `watermark.html` into the template system and it is already possible for a plugin to replace `watermark.html` with anything it likes. I saw the changes made for dark mode support and I thought it offered an opportunity to change the logo without the need for `watermark.html` to use the template system.

My thinking was that people can replace the Roundcube logo with any image they like and not only by config but also simply by replacing `logo.svg` with their own file but what looks good on a light background does not necessarily look good on a dark one and so it would be useful to be able to change the logo on the `watermark.html` screen for light/dark modes just like in the rest of Roundcube (eg. login screen vs mail screen, light vs dark backgrounds). I totally understand there are different ways of achieving this and dark mode in Roundcube is very new and still a work in progress and as things develop better approaches can emerge.

**Comment by gianlucagiacometti** [comment]
It's true that dark mode is very new, but it's a sort of standard now, which will last for a long time.
The simplest way might be adding one field to the existing skin_logo array

```
$config['skin_logo'] = array(
  "elastic:dark:*" => "/images/logo_dark.png",
  "elastic:light:*" => "/images/logo_light.png",
  "elastic:*:*[small]" => "/images/logo_small_dark and_light.png",
  ...
  );
```
Old skins can simpoly use * instead of dark or light

Moreover, when your server hosts more than one domain, it would be nice to customise logo and watermark also per domain. So a further improvement might be
```

$config['skin_logo'] = array(
  "domainA:elastic:dark:*" => "/images/logoDomainA_dark.png",
  "domainA:elastic:light:*" => "/images/logoDomainA_light.png",
  "domainB:elastic:dark:*[small]" => "/images/logoDomainB_small_dark.png",
  "*:elastic:*:*" => "/images/default_logo_for all_domains_dark_and_light.png",
  ...
  );

```

**Comment by alecpl** [comment]
I'm not yet 100% happy with current way the dark mode is implemented. E.g. with devel_mode=0 there's visible blinking effect on page load (especially on pages with forms). As for the watermark, the fact that we apply some fancy css rules on the logo in watermark.html does not make it simpler. In the past I was against making it a template, but I think this is still on the table. I don't have my preference yet, though.

**Comment by johndoh** [comment]
Your first part about the light and dark. This PR makes that possible everywhere except `watermark.html` without the need to change the existing syntax.

The second part I disagree with i think that makes the `skin_logo` option differnt to every other config option and that makes it harder to understand when such setups probably already have handling for multiple domains in place.

**Comment by alecpl** [comment]
Also, the fact that we extended `skin_logo` setting in master moves slightly the balance in the way I think about the watermark.

**Comment by johndoh** [comment]
Another approach I though of was to include a CSS file in `watermark.html` and then the logo could come from a variable in the Less, and could be customised thanks to `_variables.less` but I have not tried it yet so I'm not sure if its practial or not.

**Comment by gianlucagiacometti** [comment]
I like the idea of the css. It would allow good manipulation.
I also perfectly understand to keep the configuration as simplest as possible. Customising by domain may stay in charge of a plugin.

**Comment by alecpl** [comment]
@johndoh Looks good. For backward compatibility I'd keep the attribute name `type` instead of `logo-type`.

**Pull request merged**